### DISCUSSION TITLE
Does prune happened when running pnpm install

### DISCUSSION THREAD

**Discussion by xkn1ght** [pnpm/pnpm#4009] [Q&A]
Really curious about prune mechanism in pnpm. How it works? does it happened on every core command? etc... 

As the doc:
> If you depend on different versions of the dependency, only the files that differ are added to the store. For instance, if it has 100 files, and a new version has a change in only one of those files, pnpm update will only add 1 new file to the store, instead of cloning the entire dependency just for the singular change.

I found [issue about caf.](https://github.com/pnpm/pnpm/issues/2470) Refer to the commit, 
> pnpm prune every files that has only one hard-link.  And, by hashing every file, the result of hash function was split to dir and subpath

I am wandering if someone could tell me a brief description of how pnpm install work with prune mechanism in pnpm. thx!

**Comment by zkochan** [comment]
The global content-addressable store is not automatically pruned. If you want to prune it, run `pnpm store prune`
